projects/clr/rocclr/os/os_posix.cpp

```diff
@@ -979,6 +959,106 @@ void Os::CxaDemangle(const std::string& name, std::string* result) {
   free(demangled);
 }
 
+namespace numa {
+
+// ================================================================================================
+NumaPolicy::NumaPolicy(const uint32_t numa_node_count) :
+  node_map_((numa_node_count + kBitsPerUInt64 - 1) / kBitsPerUInt64, 0) { }
+
+// ================================================================================================
+bool NumaPolicy::GetMemPolicy() {
+  int policy = 0;
+  if (syscall(__NR_get_mempolicy, &policy, node_map_.data(),
+      node_map_.size() * kBitsPerUInt64, nullptr, 0) < 0) {
+    ClPrint(amd::LOG_DEBUG, amd::LOG_RESOURCE,
+        "syscall(__NR_get_mempolicy, size=%zu) failed to query policy",
+        node_map_.size() * kBitsPerUInt64);
+    return false;
+  }
+  if (policy < static_cast<int>(Policy::kDefault) || policy > static_cast<int>(Policy::kMax)) {
+    ClPrint(amd::LOG_DEBUG, amd::LOG_RESOURCE,
+            "syscall(__NR_get_mempolicy) returned wrong policy %d", policy);
+    return false;
+  }
+  policy_ = static_cast<Policy>(policy);
+  return true;
+}
+
+// ================================================================================================
+bool NumaPolicy::IsPolicySetAt(uint32_t node_index) const {
+  const uint32_t i = node_index / kBitsPerUInt64;
+  if (i < node_map_.size()) {
+    return ((node_map_[i] >> (node_index % kBitsPerUInt64)) & 1) ?
+        true: false;
+  } else {
+    return false;
+  }
+}
+
+// ================================================================================================
+NumaNode::~NumaNode() {
+  if (affinity_) {
+    delete static_cast<std::vector<uint64_t> *>(affinity_);
+    affinity_ = nullptr;
+  }
+}
+
+// ================================================================================================
+bool NumaNode::GetAffinity() {
+  const std::string path = "/sys/devices/system/node/node" + std::to_string(node_index_) +
+      "/cpumap";
+  std::ifstream file(path);
+  if (!file) {
+    std::cerr << "Failed to open " << path << "\n";
+    ClPrint(amd::LOG_DEBUG, amd::LOG_RESOURCE, "%s cannot be opened", path);
+    return false;
+  }
```

Comment:
This broke the build, see https://github.com/ROCm/TheRock/pull/1943 and https://github.com/ROCm/rocm-systems/pull/1576.

The fix is
```diff
-    ClPrint(amd::LOG_DEBUG, amd::LOG_RESOURCE, "%s cannot be opened", path);
+    ClPrint(amd::LOG_DEBUG, amd::LOG_RESOURCE, "%s cannot be opened", path.c_str());
```

@lamikr did you capture build logs showing the error?